### PR TITLE
[SPARK-LLAP-125][BRANCH-1.6] Reduce the number of included artifacts

### PR DESCRIPTION
**Pull request by merlintang**
…in the uber jar

## What changes were proposed in this pull request?

reduce dependency as the branch 2.1

## How was this patch tested?

I manually verify the dependence list between this patch and branch HDP-2.6.1.0-1.6

### CODE REVIEW

**Comment by merlintang** [comment]
@dongjoon-hyun  please review and verify this.

**Comment by dongjoon-hyun** [comment]
Could you put the version comparision result (before and after) into PR description?

File: build.sbt [hunk 1]
```diff
@@ -86,10 +98,24 @@ libraryDependencies ++= Seq(
...
+    .exclude("org.apache.hadoop", "hadoop-hdfs")
+    .exclude("org.apache.hbase", "*")
+    .exclude("commons-beanutils", "commons-beanutils-core")
+    .exclude("commons-collections", "commons-collections")
+    .exclude("commons-logging", "commons-logging")
 )
 dependencyOverrides += "com.google.guava" % "guava" % "16.0.1"
 dependencyOverrides += "commons-codec" % "commons-codec" % "1.6"
 dependencyOverrides += "commons-logging" % "commons-logging" % "1.2"
+dependencyOverrides += "org.apache.httpcomponents" % "httpclient" % "4.5.2"
```
Comment (dongjoon-hyun) [inline build.sbt]: I'm wondering where you bring this version, `4.5.2`.
This is not a previous version. Also, this is not a version defined Spark 1.6.

**Comment by dongjoon-hyun** [comment]
Oh, did you compare the dependecy with branch HDP-2.6.1.0-1.6 ?

**Comment by merlintang** [comment]
yes, I am.

**Comment by dongjoon-hyun** [comment]
Yep. Let me verify with that branch again. Previously, I compared with branch-1.6.

**Comment by dongjoon-hyun** [comment]
**HDP-2.6.1.0-1.6**
```
[info] org.apache.httpcomponents:httpclient:4.2.5
[info] org.apache.httpcomponents:httpcore:4.2.4
```
**branch-1.6**
```
[info] org.apache.httpcomponents:httpclient:4.4
[info] org.apache.httpcomponents:httpcore:4.4
```
??

**Comment by dongjoon-hyun** [comment]
No, the above is my result in `HDP-2.6.1.0-1.6`.

**Comment by merlintang** [comment]
What is your meaning for "the above is my result in HDP-2.6.1.0-1.6. "

Below is my comparison result:

for the branch 1.6 with the patch:

ESC[0m[ESC[0minfoESC[0m] ESC[0morg.apache.httpcomponents:httpclient:4.5.2ESC
[0m

ESC[0m[ESC[0minfoESC[0m] ESC[0morg.apache.httpcomponents:httpcore:4.4.4ESC
[0m

For the *HDP-2.6.1.0-1.6*

ESC[0m[ESC[0minfoESC[0m] ESC[0morg.apache.httpcomponents:httpclient:4.5.2ESC
[0m

ESC[0m[ESC[0minfoESC[0m] ESC[0morg.apache.httpcomponents:httpcore:4.4.4ESC
[0m

**Comment by dongjoon-hyun** [comment]
@merlintang . Could you try in a clean clone?
I think we are seeing different source code. Did you push your `HDP-2.6.1.0-1.6` to the pubilic?
The public HDP-2.6.1.0-1.6 uses 1.6.3.2.6.1.0-114. Is it latest HDP version?
```
sparkVersion := sys.props.getOrElse("spark.version", "1.6.3.2.6.1.0-114")
```

**Comment by dongjoon-hyun** [comment]
Here is my advice.
Please close this PR.
Make a PR for bump to only HDP 2.6.1 first.

**Comment by merlintang** [comment]
ok, no problem. I will create a PR toward to HDP-2.6.1.0-1.6

**Comment by dongjoon-hyun** [comment]
No. Please make a PR for branch-1.6 directly.

**Comment by merlintang** [comment]
Oh, so what is your mean "Make a PR for bump to only HDP 2.6.1 first."

**Comment by dongjoon-hyun** [comment]
See this.
https://github.com/hortonworks-spark/spark-llap/commit/c0cafa91f12900e1005aa45e1e6732d494805a86

**Comment by merlintang** [comment]
ok, I see you meaning, I at first file a PR to bump to HDP 2.6.1, then file
a new PR for artifact reducing.

**Comment by dongjoon-hyun** [comment]
Right. That will help both of us to verify your PR.
BTW, you should pass the Travis CI from now.
Usually, the existing failures are due to Scala Style error from #117. Please fix all of them.

**Comment by merlintang** [comment]
sure.

**Comment by merlintang** [comment]
I file a new PR, and close this one.

**Pull request closed without merging**